stl/inc/xstring

```diff
@@ -2931,9 +3068,10 @@ private:
         // NOTE: even if _Right is in large mode, we only go into large mode ourselves if the actual size of _Right
         // requires it
         if (_Right_size < _BUF_SIZE) { // stay small, don't allocate
-            _Traits::copy(_My_data._Bx._Buf, _Right_ptr, _BUF_SIZE);

```

Comment:
Note this is a correctness fix, that has no practical application other than making asan happy

Reply:
While it may make asan happy, this isn't a correctness fix - the previous code was also correct. This may be a micro-pessimization.

Reply:
Are you sure, if `_Right` never filled the SSO, we could copy uninitialized memory, as the underlying buffer is not value initialized

Reply:
I've overclaimed a bit here: the previous code was portably correct when `_Traits::copy` calls `memcpy`. `memcpy` is immune to indeterminate values since it copies the object representation and never performs lvalue-to-rvalue conversion on the object being copied. Presumably the `constexpr` `memcpy` intrinsics provided by compilers have the same immunity and can similarly copy indeterminate values.

In any case, I am generally opposed to pessimizing performance of runtime containers to make constexpr containers work. - this occurrence is admittedly a minor concern at best.

Reply:
Just to be sure that I understand you correctly, this reduces the number of elements we pass to `_Traits::copy`

Are you alluding to the optimizer not being able to replace this with a more efficient vector instruction? Otherwise I would not know how this would regress performance

Reply:
The optimizer has a specific case for `memcpy(a, b, a constant value)`; if you make it a non constant value the performance will be worse. Proof: https://gcc.godbolt.org/z/dGYszfh6n

1, 2, 4, 8, 16, and 32 are particularly nice because they fit in a register.

Reply:
That is unfortunate. We could in theory just make two additional calls to `__sanitizer_annotate_contiguous_memory` but I am pretty sure that is a pessimization. 

Is the benefit big enough to warrant an `#if defined(__SANITIZE_ADDRESS__) || defined(_ANNOTATE_STRING)` special casing?

Reply:
I created a single specialization of the `_Construct_from` function that takes special cases construction from ptr, char and string